# Author: Satish Kamath (email: satishs.kamath@gmail.com)
# The path may be a symbolic link to the actual paths on different nodes and the paths generated by qmake are the actual paths which are textually different. 
# The original code in the patch tries to manipulate the qtlib paths generated by qmake (the actual paths) using the symlink path (in $LIBPATH) 
# which results in wrong library paths for QT in the makefile. This has been remedied using readlink.
diff -Nru cubegui-4.6_orig/build-config/m4/autotroll.m4 cubegui-4.6/build-config/m4/autotroll.m4
--- cubegui-4.6_orig/build-config/m4/autotroll.m4	2021-08-06 13:03:13.758365694 +0200
+++ cubegui-4.6/build-config/m4/autotroll.m4	2021-08-06 13:06:18.972956958 +0200
@@ -741,7 +741,7 @@
   ])
 AS_IF([ test "x$LIBPATH" != "x" ],  # perform correction only if one has found .so libraries instead of -lXXXX
 [
-  at_cv_env_QT_LIBS="-L$LIBPATH "`echo $at_cv_env_QT_LIBS | sed "s,.so , ,g;s,.a , ,g;s,$LIBPATH/lib,-l,g"`
+  at_cv_env_QT_LIBS="-L$LIBPATH "`echo $at_cv_env_QT_LIBS | sed "s,.so , ,g;s,.a , ,g;s,$(readlink -m $LIBPATH)/lib,-l,g"`
   echo "Fixed libraries are: $at_cv_env_QT_LIBS"
 ])
   AC_SUBST([QT_LIBS], [$at_cv_env_QT_LIBS])
